### PR TITLE
sg: skip honey event duration if event is nil

### PR DESCRIPTION
**Pull request by burmudar**
`runScripSerialized` worked fine if you ran `pnpm dedupe` because it was 'instrumented' with `timeCheck` but if you didn't and executed `runScriptSerialized` without `timeCheck` the context won't be Honeycomb enriched and thus the event will be nil and then the whole method falls apart.

Previously this also just checked the locking duration of the mutex 🤔 and I assume it wanted to check the duration of the `pnpm dedupe`?

Either way, it now checks if the event is nil before using it.

I think this addresses #62471 and https://github.com/sourcegraph/devx-support/issues/933
## Test plan
Tested locally

### CODE REVIEW

File: dev/sg/linters/linters.go [hunk 1]
```diff
@@ -133,6 +133,11 @@ func runScript(name string, script string) *linter {
 	return &linter{
 		Name: name,
 		Check: func(ctx context.Context, out *std.Output, args *repo.State) error {
+			defer func() {
```
Comment (Strum355) [inline dev/sg/linters/linters.go:136]: TODO: rm

**Comment by Strum355** [comment]
Not sure if we even still need any of this 🤔 we could maybe delete this as we dont check honeycomb stats for sg lint anymore

File: dev/sg/linters/linters.go (hunk 1 above)
Comment (burmudar) [reply dev/sg/linters/linters.go:136]: :wat: I thought I remove all of this 🤣

**Comment by burmudar** [comment]
> Not sure if we even still need any of this 🤔 we could maybe delete this as we dont check honeycomb stats for sg lint anymore

yeah then I'll remove it

File: dev/sg/linters/linters.go [hunk 2]
```diff
@@ -149,14 +149,19 @@ func runScriptSerialized(name string, script string) *linter {
...
-			event := honey.FromContext(ctx)
-
 			t1 := time.Now()
 			runScriptSerializedMu.Lock()
-			t2 := time.Since(t1)
-			event.AddField("pnpm_lock_duration", t2.Seconds())
-			event.AddField("pnpm_lock_duration_ms", t2.Milliseconds())
 			defer runScriptSerializedMu.Unlock()
+			// record the time it took to run the script, if it does have a honey event
+			defer func() {
```
Comment (burmudar) [inline dev/sg/linters/linters.go:156]: need to remove this

File: dev/sg/linters/linters.go [hunk 3]
```diff
@@ -61,42 +59,42 @@ var Targets = []Target{
...
-			timeCheck(checkUnversionedDocsLinks()),
+			checkUnversionedDocsLinks(),
 		},
 	},
 	{
 		Name:        "pnpm",
 		Description: "Check pnpm lockfiles for optimality",
 		Checks: []*linter{
-			timeCheck(runScriptSerialized("pnpm dedupe", "dev/check/pnpm-deduplicate.sh")),
+			runScriptSerialized("pnm dedupe", "dev/check/pnpm-deduplicate.sh"),
```
Comment (Strum355) [inline dev/sg/linters/linters.go:79]:
```suggestion
			runScriptSerialized("pnpm dedupe", "dev/check/pnpm-deduplicate.sh"),
```

**Review by Strum355 — APPROVED** [review]

File: dev/sg/linters/linters.go (hunk 3 above)
Comment (burmudar) [reply dev/sg/linters/linters.go]: Great catch

**Pull request merged**